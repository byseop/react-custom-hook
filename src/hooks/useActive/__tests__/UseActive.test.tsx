import React from 'react';
import UseActiveExmaple from '../UseActiveExample';
import { render, fireEvent } from '@testing-library/react';

describe('<UseActiveExample />', () => {
  it('matches snapshot', () => {
    const utils = render(<UseActiveExmaple />);
    expect(utils.container).toMatchSnapshot();
  });

  it('shows the props correctly', () => {
    const utils = render(<UseActiveExmaple />);
    utils.getByText('Active');
    utils.getByText('Deactive');
    utils.getByText('Deactivate');
  });

  it('click activate button', () => {
    const utils = render(<UseActiveExmaple />);
    const isActive = utils.getByText('Deactivate');
    const acivateButton = utils.getByText('Active');

    fireEvent.click(acivateButton);

    expect(isActive).toHaveTextContent('Activate');
    expect(isActive.textContent).toBe('Activate');
  });

  it('click deactivate button', () => {
    const utils = render(<UseActiveExmaple />);
    const isActive = utils.getByText('Deactivate');
    const acivateButton = utils.getByText('Active');
    const deactivateButton = utils.getByText('Deactive');

    fireEvent.click(acivateButton);
    fireEvent.click(deactivateButton);
    
    expect(isActive).toHaveTextContent('Deactivate');
    expect(isActive.textContent).toBe('Deactivate');
  });
});
