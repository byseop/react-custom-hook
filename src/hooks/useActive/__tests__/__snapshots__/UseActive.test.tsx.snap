// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UseActiveExample /> matches snapshot 1`] = `
<div>
  <div>
    Deactivate
  </div>
  <button>
    Active
  </button>
  <button>
    Deactive
  </button>
</div>
`;
